streams-core/src/main/java/org/apache/streams/core/StreamsDatum.java

```diff
@@ -51,10 +51,22 @@ public StreamsDatum(Object document, DateTime timestamp, BigInteger sequenceid)
         this(document, null, timestamp, sequenceid);
     }
 
+    public StreamsDatum(Object document, DateTime timestamp, Map<String, Object> metadata) {
+        this(document, null, timestamp, null, metadata);
+    }
+
     public StreamsDatum(Object document, String id, DateTime timestamp) {
         this(document, id, timestamp, null);
```

Comment:
If this is calling the constructor on line 66, then passing null for metadata is inconstant with the empty hash in the constructors on 34 and 38.